resources/splunk_service.rb

```diff
@@ -62,14 +65,30 @@ def install_state
   action_class do
     include CernerSplunk::ProviderHelpers
 
-    def service_action(desired_action)
+    # Provide file and service resources from methods to eliminate accidental resource cloning
+    def marker_resource
+      resources(file: marker_path.to_s)
+    rescue Chef::Exceptions::ResourceNotFound
+      file(marker_path.to_s) do
+        action :nothing
+      end
+    end
+
+    def service_resource
+      resources(service: service_name)
+    rescue Chef::Exceptions::ResourceNotFound
       service service_name do
         provider Chef::Provider::Service::Systemd if systemd_is_init?
         supports start: true, stop: true, restart: true, status: true
-        action desired_action
+        action :nothing
       end
     end
 
+    def service_action(desired_action)
+      service_resource.run_action(desired_action)
+      marker_resource.run_action(:delete)
```

Comment:
You're right... I came to this implementation trying to fix resource cloning but I forgot that requirement.